drivers/osdp/src/osdp_pd.c

```diff
@@ -0,0 +1,576 @@
+			max_len -= 3;
+		}
+		break;
+	case REPLY_LSTATR:
+		if (max_len < 2) {
+			LOG_ERR(TAG "Out of buffer space!");
+			return -1;
+		}
+		buf[len++] = isset_flag(pd, PD_FLAG_TAMPER);
+		buf[len++] = isset_flag(pd, PD_FLAG_POWER);
+		break;
+	case REPLY_RSTATR:
+		if (max_len < 1) {
+			LOG_ERR(TAG "Out of buffer space!");
+			return -1;
+		}
+		buf[len++] = isset_flag(pd, PD_FLAG_R_TAMPER);
+		break;
+	case REPLY_COM:
+		if (max_len < 5) {
+			LOG_ERR(TAG "Out of buffer space!");
+			return -1;
+		}
+		buf[len++] = pd->address;
+		buf[len++] = byte_0(pd->baud_rate);
+		buf[len++] = byte_1(pd->baud_rate);
+		buf[len++] = byte_2(pd->baud_rate);
+		buf[len++] = byte_3(pd->baud_rate);
+		break;
+	case REPLY_NAK:
+		if (max_len < 1) {
+			LOG_ERR(TAG "Out of buffer space!");
+			return -1;
+		}
+		buf[len++] = reply->cmd_bytes[0];
+		break;
+	}
+
+	if (smb && (smb[1] > SCS_14) && isset_flag(pd, PD_FLAG_SC_ACTIVE)) {
+		smb[0] = 2;
+		smb[1] = (len > 1) ? SCS_18 : SCS_16;
+	}
+
+	if (len == 0) {
+		buf[len++] = REPLY_NAK;
+		buf[len++] = OSDP_PD_NAK_SC_UNSUP;
+	}
+
+	return len;
+}
+
+/**
+ * pd_send_reply - blocking send; doesn't handle partials
+ * Returns:
+ *   0 - success
+ *  -1 - failure
+ */
+int pd_send_reply(struct osdp_pd *pd, struct osdp_cmd *reply)
+{
+	int ret, len;
+
+	/* init packet buf with header */
+	len = phy_build_packet_head(pd, reply->id, pd->rx_buf,
+				    sizeof(pd->rx_buf));
+	if (len < 0) {
+		LOG_ERR(TAG "Failed to build head; ID: %02X", reply->id);
+		return -1;
+	}
+
+	/* fill reply data */
+	ret = pd_build_reply(pd, reply, pd->rx_buf, sizeof(pd->rx_buf));
+	if (ret <= 0) {
+		LOG_ERR(TAG "Failed build reply; ID: %02X", reply->id);
+		return -1;
+	}
+	len += ret;
+
+	/* finalize packet */
+	len = phy_build_packet_tail(pd,  pd->rx_buf, len, sizeof(pd->rx_buf));
+	if (len < 0) {
+		LOG_ERR(TAG "Failed to finalize reply; ID: %02X", reply->id);
+		return -1;
+	}
+
+	ret = pd->channel.send(pd->channel.data, pd->rx_buf, len);
+
+#ifdef CONFIG_OSDP_PACKET_TRACE
+	if (pd->cmd_id != CMD_POLL) {
+		osdp_dump("PD sent", pd->rx_buf, len);
+	}
+#endif
+
+	return (ret == len) ? 0 : -1;
+}
+
+/**
+ * pd_receve_packet - received buffer from serial stream handling partials
+ * Returns:
+ *  0: success
+ * -1: error
+ *  1: no data yet
+ */
+int pd_receve_packet(struct osdp_pd *pd)
+{
+	uint8_t *buf;
+	int ret, new_packet, max_len;
+
+	new_packet = pd->rx_buf_len == 0;
+	buf = pd->rx_buf + pd->rx_buf_len;
+	max_len = sizeof(pd->rx_buf) - pd->rx_buf_len;
+
+	ret = pd->channel.recv(pd->channel.data, buf, max_len);
+	if (ret <= 0)
+		return 1;
+	if (new_packet && ret > 0)
+		pd->tstamp = millis_now();
+	pd->rx_buf_len += ret;
+
+#ifdef CONFIG_OSDP_PACKET_TRACE
+	if (pd->rx_buf[6] != CMD_POLL && pd->rx_buf[8] != CMD_POLL) {
+		osdp_dump("PD received", pd->rx_buf, pd->rx_buf_len);
+	}
+#endif
+
+	ret = phy_decode_packet(pd, pd->rx_buf, pd->rx_buf_len);
+	if (ret < 0) {
+		switch (ret) {
+		case -1: /* fatal error */
```

Comment:
Could you enums/defines for the error values, using plain values should be avoided

Reply:
Done.